protocol/x/listing/keeper/listing_test.go

```diff
@@ -0,0 +1,188 @@
+package keeper_test
+
+import (
+	"errors"
+	"testing"
+
+	perpetualtypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	oracletypes "github.com/skip-mev/slinky/pkg/types"
+	marketmaptypes "github.com/skip-mev/slinky/x/marketmap/types"
+	"github.com/skip-mev/slinky/x/marketmap/types/tickermetadata"
+
+	pricestypes "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+
+	"github.com/dydxprotocol/v4-chain/protocol/x/listing/types"
+
+	"github.com/dydxprotocol/v4-chain/protocol/mocks"
+	keepertest "github.com/dydxprotocol/v4-chain/protocol/testutil/keeper"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateMarket(t *testing.T) {
+	tests := map[string]struct {
+		ticker          string
+		duplicateMarket bool
+
+		expectedErr error
+	}{
+		"success": {
+			ticker:          "TEST-USD",
+			duplicateMarket: false,
+			expectedErr:     nil,
+		},
+		"failure - invalid market": {
+			ticker:          "INVALID-USD",
+			duplicateMarket: false,
+			expectedErr:     types.ErrMarketNotFound,
+		},
+		"failure - duplicate market": {
+			ticker:          "TEST-USD",
+			duplicateMarket: true,
+			expectedErr:     nil,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(
+			name, func(t *testing.T) {
+				mockIndexerEventManager := &mocks.IndexerEventManager{}
+				ctx, keeper, _, _, pricesKeeper, _, _, marketMapKeeper := keepertest.ListingKeepers(
+					t,
+					&mocks.BankKeeper{},
+					mockIndexerEventManager,
+				)
+
+				testMarketParams := pricestypes.MarketParam{
+					Pair:               "TEST-USD",
+					Exponent:           int32(-6),
+					ExchangeConfigJson: `{"test_config_placeholder":{}}`,
+					MinExchanges:       2,
+					MinPriceChangePpm:  uint32(800),
+				}
+
+				keepertest.CreateMarketsInMarketMapFromParams(
+					t,
+					ctx,
+					marketMapKeeper,
+					[]pricestypes.MarketParam{
+						testMarketParams,
+					},
+				)
+
+				marketId, err := keeper.CreateMarket(ctx, tc.ticker)
+				if tc.expectedErr != nil {
+					require.Error(t, err)
+				} else {
+					require.NoError(t, err)
+
+					// Check if the market was created
+					market, exists := pricesKeeper.GetMarketParam(ctx, marketId)
+					require.True(t, exists)
+					require.Equal(t, testMarketParams.Pair, market.Pair)
+					require.Equal(t, testMarketParams.Exponent, market.Exponent)
+					require.Equal(t, testMarketParams.MinExchanges, market.MinExchanges)
+					require.Equal(t, testMarketParams.MinPriceChangePpm, types.MinPriceChangePpm_LongTail)
+				}
+
+				if tc.duplicateMarket {
+					_, err = keeper.CreateMarket(ctx, tc.ticker)
+					require.ErrorContains(t, err, pricestypes.ErrMarketParamPairAlreadyExists.Error())
+				}
```

Comment:
non-blocking nit: Test-pattern is a bit weird. Would be better to have some default market set up, and we just check that the expected error is the already exists error.